dali/test/python/test_operator_laplacian.py

```diff
@@ -0,0 +1,573 @@
+                           ((20, 30, 3), "HWC", 2), ((20, 30), "HW", 2),
+                           ((3, 30, 20), "CWH", 2), ((5, 20, 30, 3), "FHWC", 2),
+                           ((5, 10, 10, 7, 3), "FDHWC", 3),
+                           ((5, 3, 20, 30), "FCHW", 2),
+                           ((3, 5, 10, 10, 7), "FCDHW", 3)]
+
+
+def to_batch(tl, batch_size):
+    return [np.array(tl[i]) for i in range(batch_size)]
+
+
+# Simple check if laplacian of a square matrix that has zeros everywhere except
+# the middle cell, gives sum of separable convolution kernels used to compute
+# the partial derivatives, i.e. sum of the products of 1D convolution windows.
+def _test_kernels(num_dims, smoothing, normalize):
+    batch_size = (max_window_size + 2 - min_window_size) // 2
+
+    def get_inputs():
+        ones = []
+        window_sizes = []
+        smoothing_sizes = []
+        scales = []
+        padding = 2
+        for k in range(min_window_size, max_window_size + 2, 2):
+            a_size = k + padding
+            a = np.full((a_size,) * num_dims, 0, dtype=np.float32)
+            a[(a_size // 2,) * num_dims] = 1
+            ones.append(a)
+            window_sizes.append(np.array(k, dtype=np.int32))
+            if smoothing:
+                smoothing_sizes.append(np.array(k, dtype=np.int32))
+                exponent = num_dims * k - 2 - num_dims
+            else:
+                smoothing_sizes.append(np.array(1, dtype=np.int32))
+                exponent = k - 3
+            scales.append(np.array(2.**(-exponent), dtype=np.float32))
+        return ones, window_sizes, smoothing_sizes, scales
+
+    @pipeline_def
+    def pipeline():
+        ones, window_sizes, smoothing_sizes, scales = fn.external_source(
+            get_inputs, num_outputs=4)
+        kernels = fn.laplacian(
+            ones, window_size=window_sizes, smoothing_size=smoothing_sizes,
+            dtype=types.FLOAT, normalize=normalize)
+        return kernels, scales
+
+    def outer(*vs):
+        acc = np.array([1.])
+        for v in vs:
+            acc = np.outer(acc, v)
+        return acc.reshape(tuple(len(v) for v in vs))
+
+    def get_cv2_kernel(k, smoothing):
+        d, s = cv2.getDerivKernels(2, 0, k)
+        if not smoothing:
+            s = np.zeros(k)
+            s[k // 2] = 1.
+        windows = [[d if i == j else s for j in range(
+            num_dims)] for i in range(num_dims)]
+        return sum(outer(*ws) for ws in windows)
+
+    pipe = pipeline(num_threads=4, batch_size=batch_size,
+                    device_id=types.CPU_ONLY_DEVICE_ID)
+    pipe.build()
+    (kernels, scales) = pipe.run()
+    kernels = [np.array(ker)[(slice(1, -1),) * num_dims] for ker in kernels]
+    scales = [np.array(sf).item() for sf in scales]
+    rng = range(min_window_size, max_window_size + 2, 2)
+    assert (len(kernels) == len(rng) == len(scales))
+    for scale_factor, kernel, k in zip(scales, kernels, rng):
```

Comment:
This `k` confuses me, How about `w` like window?

Reply:
Just renamed it to `win_size` everywhere.